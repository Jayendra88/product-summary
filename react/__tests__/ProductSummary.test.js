import React from 'react'
import { render } from 'test-utils'

import ProductSummary from '../index'

describe('<ProductSummary /> component', () => {
  function renderComponent(customProps) {
    const props = {
      ...customProps,
      runtime: { hints: {} },
      product: {
        productId: '12345',
        linkText: 'linkText',
        productName: 'productName',
        sku: {
          name: 'name',
          itemId: 'itemId',
          image: {
            imageUrl: '',
          },
          seller: {
            sellerId: 'sellerId',
            commertialOffer: {
              Installments: [
                {
                  Value: 1,
                  InterestRate: 1,
                  NumberOfInstallments: 1,
                },
              ],
              Price: 1,
              ListPrice: 1,
            },
          },
        },
        productClusters: [
          {
            name: 'name',
          },
        ],
        quantity: 1,
      },
    }

    return render(<ProductSummary {...props} />)
  }

  it('should match the snapshot for normal mode', () => {
    const { asFragment } = renderComponent({ displayMode: 'normal' })
    expect(asFragment()).toMatchSnapshot()
  })

  it('should match the snapshot for small mode', () => {
    const { asFragment } = renderComponent({ displayMode: 'small' })
    expect(asFragment()).toMatchSnapshot()
  })

  it('should match the snapshot for inline normal mode', () => {
    const { asFragment } = renderComponent({ displayMode: 'inline' })
    expect(asFragment()).toMatchSnapshot()
  })

  it('should match the snapshot for inline price mode', () => {
    const { asFragment } = renderComponent({ displayMode: 'inlinePrice' })
    expect(asFragment()).toMatchSnapshot()
  })
})
