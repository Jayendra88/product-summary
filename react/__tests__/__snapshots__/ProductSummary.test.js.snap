// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductSummary /> component should match the snapshot for inline normal mode 1`] = `
<DocumentFragment>
  <section
    class="container containerInlinePrice overflow-hidden br3 h-100 w-100"
  >
    <article
      class="flex"
    >
      <div
        class="imageContainer db w-70"
      >
        <div>
          ContentLoader
        </div>
      </div>
      <div
        class="information w-80 pb2 pl3 flex flex-wrap flex-column justify-between"
      >
        <div
          class="flex flex-column"
        >
          <div
            class="flex items-start justify-left tl w-90 t-mini pb2"
          >
            <div>
              ProductName
            </div>
          </div>
          <div
            class="nr2"
          >
            <div
              class="flex flex-column nr1 priceContainer"
            >
              <div>
                ProductPrice
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex flex-column-reverse"
        >
          <div
            class="buyButtonContainer pt3 w-100"
          >
            <div
              class="buyButton center mw-100"
            >
              <div>
                BuyButton
              </div>
            </div>
          </div>
        </div>
      </div>
    </article>
  </section>
</DocumentFragment>
`;

exports[`<ProductSummary /> component should match the snapshot for inline price mode 1`] = `
<DocumentFragment>
  <section
    class="container containerInline overflow-hidden br3 h-100 w-100"
  >
    <article
      class="element pointer ph2 pt3 pb4 flex flex-column"
    >
      <div
        class="imageContainer db w-30"
      >
        <div>
          ContentLoader
        </div>
      </div>
      <div
        class="information w-70 pb2 pl3 pr3"
      >
        <div
          class="flex items-start justify-left tl w-90 t-mini pb2"
        >
          <div>
            ProductName
          </div>
        </div>
        <div
          class="mt3 nr2"
        >
          <div
            class="flex justify-end nr4 mb2"
          >
            <div>
              NumericStepper
            </div>
          </div>
          <div
            class="flex flex-column items-end nr1 h1 priceContainer pv5"
          >
            <div>
              ProductPrice
            </div>
          </div>
        </div>
      </div>
      <div
        class="buyButtonContainer pv3 w-100 dn"
      >
        <div
          class="buyButton center mw-100"
        >
          <div>
            BuyButton
          </div>
        </div>
      </div>
    </article>
  </section>
</DocumentFragment>
`;

exports[`<ProductSummary /> component should match the snapshot for normal mode 1`] = `
<DocumentFragment>
  <section
    class="container containerNormal overflow-hidden br3 h-100 flex flex-column justify-between center tc"
  >
    <div
      class="absolute z-1 mt3"
    >
      <div
        id="wish-list"
        product="[object Object]"
      />
    </div>
    <article
      class="element pointer pt3 pb4 flex flex-column h-100"
    >
      <div
        class="imageContainer db w-100 center"
      >
        <div>
          ContentLoader
        </div>
      </div>
      <div
        class="information h-100 flex flex-column justify-between"
      >
        <div
          class="flex items-start nameContainer justify-center pv6"
        >
          <div>
            ProductName
          </div>
        </div>
        <div>
          <div
            class="flex flex-column justify-end items-center relative priceContainer pv5"
          >
            <div>
              ProductPrice
            </div>
          </div>
        </div>
      </div>
    </article>
    <div
      class="buyButtonContainer pv3 w-100 db"
    >
      <div
        class="buyButton center mw-100"
      >
        <div>
          BuyButton
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`<ProductSummary /> component should match the snapshot for small mode 1`] = `
<DocumentFragment>
  <section
    class="container containerSmall overflow-hidden br3 h-100 flex flex-column justify-between center tc"
  >
    <article
      class="element pointer pt3 pb4 flex flex-column"
    >
      <div
        class="imageContainer db w-100 center"
      >
        <div>
          ContentLoader
        </div>
      </div>
      <div
        class="information"
      >
        <div
          class="flex items-start nameContainer justify-center pv5 t-mini pb2"
        >
          <div>
            ProductName
          </div>
        </div>
        <div>
          <div
            class="flex flex-column justify-end items-center priceContainer pv5"
          >
            <div>
              ProductPrice
            </div>
          </div>
        </div>
      </div>
    </article>
    <div
      class="buyButtonContainer pv3 w-100 dn"
    >
      <div
        class="buyButton center mw-100"
      >
        <div>
          BuyButton
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
